### ISSUE TITLE
Data Overlap Problem

### ISSUE THREAD

**Issue by 01-hyeok** [thuml/iTransformer#162]
Dear Author,
I appreciate your excellent work on iTransformer and its application to time-series forecasting. However, I noticed a potential issue in the way training, validation, and test sets are split in your implementation.

In the dataset processing code, the boundary indices for train, validation, and test sets are defined as follows:

```
border1s = [0, num_train - self.seq_len, len(df_raw) - num_test - self.seq_len]
border2s = [num_train, num_train + num_vali, len(df_raw)]
```

This approach subtracts seq_len from the start of the validation set (num_train - seq_len). However, this can cause an overlap between the training and validation sets, as the last seq_x in the training set may include the same data points as the first seq_x in the validation set.

So I wonder if you intended these things.

**Comment by WenWeiTHU** [comment]
Thanks for your questions. There are some prior discussions about this. I hope this can be helpful to you.
* https://github.com/thuml/Autoformer/issues/20
* https://github.com/zhouhaoyi/Informer2020/issues/164

**Comment by 01-hyeok** [comment]
Thank you for your help!
